pmd-core/src/main/java/net/sourceforge/pmd/SynchronizedReportListener.java

```diff
@@ -9,7 +9,7 @@
 /**
  * Wraps a report listener in order to synchronize calls to it.
  */
-public final class SynchronizedReportListener implements ReportListener {
+public final class SynchronizedReportListener implements ThreadSafeReportListener {
```

Comment:
Do we still need this `SynchronizedReportListener`? I think, PMD itself doesn't use it directly anymore with your changes... so we could delete it (or deprecate it)

Or should we provide it as an example of a very simple, but possibly too much synchronizing listener, e.g. as a help to migrate your existing listener to a `ThreadSafeReportListener`?